ckb-debugger-tests/src/lib.rs

```diff
@@ -95,3 +97,130 @@ impl From<packed::Script> for ChildScript {
         ChildScript::new_unchecked(value.as_bytes())
     }
 }
+
+// Now, only support lock script
+fn get_index_by_group_index(group_index: usize, repr_tx: &ReprMockTransaction) -> Vec<usize> {
+    let mut r = Vec::<usize>::new();
+
+    let mut group_index = group_index;
+    let mut hash_buf = HashSet::<Byte32>::new();
+
+    let mut cur_script_hash: Option<Byte32> = None;
+
+    for i in 0..repr_tx.mock_info.inputs.len() {
+        let input_cell = repr_tx.mock_info.inputs.get(i).unwrap();
+        let hash = {
+            let script = &input_cell.output.lock;
+            let script = Script::new_builder()
+                .args(script.args.as_bytes().pack())
+                .code_hash(script.code_hash.pack())
+                .hash_type((script.hash_type.clone() as u8).into())
+                .build();
+            script.calc_script_hash()
+        };
+
+        if group_index == 0 {
+            if cur_script_hash.is_none() {
+                cur_script_hash = Some(hash.clone());
+                r.push(i);
+            } else {
+                if cur_script_hash.eq(&Some(hash.clone())) {
+                    r.push(i);
+                }
+            }
+        } else if hash_buf.get(&hash).is_none() {
+            hash_buf.insert(hash.clone());
+            group_index -= 1;
+        }
+    }
+
+    r
+}
+
+pub fn generate_sighash_all(
+    tx: &MockTransaction,
+    group_index: usize,
+) -> Result<[u8; 32], anyhow::Error> {
+    let repr_tx: ReprMockTransaction = tx.clone().into();
+
+    let lock_indexs = get_index_by_group_index(group_index, &repr_tx);
+    if lock_indexs.is_empty() {
+        panic!("not get lock index");
+    }
+
+    let zero_extra_witness = {
+        let mut buf = Vec::new();
+        let witness = tx
+            .tx
+            .witnesses()
+            .get(lock_indexs[0])
+            .expect("index:0 witness");
+        buf.resize(witness.len() - 20, 0);
```

Comment:
the size should be length of `lock`. Think about the case: `input_type` and `output_type` is not none